### PR TITLE
Add attachment flags & voice message fields

### PR DESCRIPTION
**Pull request by LaptopCat**
also 2 attachment options related to voice messages

### CODE REVIEW

File: pages/resources/message.mdx [hunk 1]
```diff
@@ -449,24 +449,37 @@ For the `attachments` array in Message Create/Edit requests, only the `id` is re
...
+| uploaded_filename? ^2^ | string    | The name of the file pre-uploaded to Discord's GCP bucket                               |
+| description?           | string    | The description for the file (max 1024 characters)                                      |
+| content_type?          | string    | The attachment's [media type](https://en.wikipedia.org/wiki/Media_type)                 |
+| size                   | integer   | The size of file in bytes                                                               |
+| url                    | string    | Source URL of the file                                                                  |
+| proxy_url              | string    | A proxied url of the file                                                               |
+| height?                | ?integer  | Height of file (if image)                                                               |
+| width?                 | ?integer  | Width of file (if image)                                                                |
+| ephemeral? ^1^         | boolean   | whether this attachment is ephemeral                                                    |
```
Comment (splatterxl) [inline pages/resources/message.mdx:464]:
```suggestion
| ephemeral? ^1^         | boolean   | Whether this attachment is ephemeral                                                    |
```

While you're modifying this section why not update this for consistency

File: pages/resources/message.mdx [hunk 2]
```diff
@@ -449,24 +449,37 @@ For the `attachments` array in Message Create/Edit requests, only the `id` is re
...
 
 ^1^ Ephemeral attachments will automatically be removed after a set period of time. Ephemeral attachments on messages are guaranteed to be available as long as the message itself exists.
 
 ^2^ This field is send-only. See [Uploading to Google Cloud](/reference#uploading-to-google-cloud) for more information.
 
+###### Attachment Flags
+
+| Value   | Name                    | Description                                                       |
+| ------- | ----------------------- | ----------------------------------------------------------------- |
+| 1 << 0  | IS_CLIP                 | This attachment is a clip                                         |
```
Comment (splatterxl) [inline pages/resources/message.mdx:477]:
```suggestion
| 1 << 0  | IS_CLIP                 | This attachment is a clipped recording of a stream                                         |
```

File: pages/resources/message.mdx [hunk 3]
```diff
@@ -449,24 +449,37 @@ For the `attachments` array in Message Create/Edit requests, only the `id` is re
...
+| content_type?          | string    | The attachment's [media type](https://en.wikipedia.org/wiki/Media_type)                 |
+| size                   | integer   | The size of file in bytes                                                               |
+| url                    | string    | Source URL of the file                                                                  |
+| proxy_url              | string    | A proxied url of the file                                                               |
+| height?                | ?integer  | Height of file (if image)                                                               |
+| width?                 | ?integer  | Width of file (if image)                                                                |
+| ephemeral? ^1^         | boolean   | whether this attachment is ephemeral                                                    |
+| duration_secs?         | float     | Duration of the audio file (currently for voice messages)                               |
+| waveform?              | string    | Base64 encoded bytearray representing a sampled waveform (currently for voice messages) |
```
Comment (dolfies) [inline pages/resources/message.mdx]: Keeping consistency with height and width
```suggestion
| duration_secs?         | float     | Duration of the audio file (if voice message)                                           |
| waveform?              | string    | Base64 encoded bytearray representing a sampled waveform (if voice message)             |
```

**Review by splatterxl — CHANGES_REQUESTED** [review]

File: pages/resources/message.mdx (hunk 1 above)
Comment (LaptopCat) [reply pages/resources/message.mdx]: Done

File: pages/resources/message.mdx [hunk 4]
```diff
@@ -449,24 +449,37 @@ For the `attachments` array in Message Create/Edit requests, only the `id` is re
...
 
+###### Attachment Flags
+
+| Value   | Name                    | Description                                                       |
+| ------- | ----------------------- | ----------------------------------------------------------------- |
+| 1 << 0  | IS_CLIP                 | This attachment is a clip                                         |
+| 1 << 1  | IS_THUMBNAIL            | This attachment is a thumbnail                                    |
+| 1 << 2  | IS_REMIX                | This attachment has been edited using the remix feature on mobile |
+| 1 << 3  | IS_SPOILER              | This attachment is a spoiler                                      |
+| 1 << 4  | CONTAINS_EXPLICIT_MEDIA | This attachment contains explicit media                           |
```
Comment (dolfies) [inline pages/resources/message.mdx]: We've made the decision to remove the IS_ prefix from flags as it's redundant :)
```suggestion
| Value  | Name                    | Description                                                                                                              |
|--------|-------------------------|--------------------------------------------------------------------------------------------------------------------------|
| 1 << 0 | CLIP                    | This attachment is a [clipped recording of a stream](https://support.discord.com/hc/en-us/articles/16861982215703-Clips) |
| 1 << 1 | THUMBNAIL               | This attachment is a thumbnail                                                                                           |
| 1 << 2 | REMIX                   | This attachment has been [remixed](https://support.discord.com/hc/en-us/articles/15145601963031-Remix-FAQ)               |
| 1 << 3 | SPOILER                 | This attachment is a spoiler                                                                                             |
| 1 << 4 | CONTAINS_EXPLICIT_MEDIA | This attachment contains explicit media                                                                                  |
```

File: pages/resources/message.mdx (hunk 2 above)
Comment (LaptopCat) [reply pages/resources/message.mdx]: Done, thank you

**Review by dolfies — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just a couple of nitpicks

**Review by dolfies — APPROVED** [review]
Thanks!

**Pull request merged**